[Bug]: ApiError module surface mismatch between @/lib/apiClient and @/lib/api/error

### Summary

`ApiError` was imported inconsistently from two different module paths: `@/lib/api/error` and `@/lib/apiClient`.

The root problem was that `@/lib/apiClient` used `ApiError` internally but did not originally export it, while several pages imported `ApiError` from `@/lib/apiClient` anyway. This created a module surface mismatch and caused TypeScript to fail with:

Module '"@/lib/apiClient"' declares 'ApiError' locally, but it is not exported.

This is a developer-facing bug that breaks type checking and can block builds or local development.


### Steps to reproduce

1. Open the frontend repository on a commit where `@/lib/apiClient` does not export `ApiError`.
2. Ensure at least one page imports `ApiError` from `@/lib/apiClient`, for example:
   - `app/(protected)/keys/page.tsx`
   - `app/(protected)/keys/[id]/page.tsx`
   - `app/(protected)/keys/[id]/return/page.tsx`
   - `app/(protected)/visits/page.tsx`
   - `app/(protected)/visits/[visitId]/visit-detail-page.tsx`
3. Run the TypeScript check:
   `npx tsc --noEmit`
4. Observe the compiler error about `ApiError` being declared locally in `@/lib/apiClient` but not exported.


### Expected behavior

`ApiError` should have a single clear owner module and all imports should resolve consistently.

TypeScript should complete without module export/import errors, and `apiClient` should not expose error types unless that is an intentional part of its public API.


### Actual behavior

The codebase mixed two import paths for the same error type:
- `@/lib/api/error`
- `@/lib/apiClient`

Because `@/lib/apiClient` did not export `ApiError`, TypeScript failed with a module export error. This made the API layer contract ambiguous and caused avoidable compile-time failures.


### Environment

- OS: Windows 11
- Shell: PowerShell
- Runtime target in repo: Node.js >=24.0.0
- TypeScript: from project devDependencies (`^5`)
- Next.js: 16.2.3
- React: 19.2.4
- App version / commit: reproduce on any commit where `ApiError` is imported from `@/lib/apiClient` but not exported there


### Relevant logs or screenshots

```shell
npx tsc --noEmit

error TS2459: Module '"@/lib/apiClient"' declares 'ApiError' locally, but it is not exported.
```

### Checklist

- [x] I searched existing issues before opening this report.
- [x] I can reproduce this issue consistently.
- [x] I understand that sensitive security issues must not be reported publicly.

--- a/app/(protected)/keys/[id]/page.tsx
+++ b/app/(protected)/keys/[id]/page.tsx
@@ -12,7 +12,7 @@ import HistoryIcon from "@mui/icons-material/History";
 import EventAvailableIcon from "@mui/icons-material/EventAvailable";
 import AssignmentReturnIcon from "@mui/icons-material/AssignmentReturn";
 import { Button } from "@/components/ui/button";
-import { ApiError } from "@/lib/apiClient";
+import { ApiError } from "@/lib/api/error";
 import {
   getKeycard,
   getKeycardStatusLabel,

--- a/app/(protected)/keys/[id]/return/page.tsx
+++ b/app/(protected)/keys/[id]/return/page.tsx
@@ -14,7 +14,7 @@ import {
   getAccessPoints,
   type AccessPointResponse,
 } from "@/lib/api/accessPoints";
-import { ApiError } from "@/lib/apiClient";
+import { ApiError } from "@/lib/api/error";
 import {
   getKeycard,
   getKeycardStatusLabel,

--- a/app/(protected)/keys/page.tsx
+++ b/app/(protected)/keys/page.tsx
@@ -14,7 +14,7 @@ import MeetingRoomIcon from "@mui/icons-material/MeetingRoom";
 import InfoIcon from "@mui/icons-material/Info";
 import { ChevronDownIcon, ChevronUpIcon } from "lucide-react";
 import { Button } from "@/components/ui/button";
-import { ApiError } from "@/lib/apiClient";
+import { ApiError } from "@/lib/api/error";
 import {
   getKeycard,
   getKeycards,

--- a/app/(protected)/visits/[visitId]/visit-detail-page.tsx
+++ b/app/(protected)/visits/[visitId]/visit-detail-page.tsx
@@ -24,7 +24,7 @@ import MeetingRoomOutlinedIcon from "@mui/icons-material/MeetingRoomOutlined";
 import PersonOutlineOutlinedIcon from "@mui/icons-material/PersonOutlineOutlined";
 import PrintOutlinedIcon from "@mui/icons-material/PrintOutlined";
 import { Button } from "@/components/ui/button";
-import { ApiError } from "@/lib/apiClient";
+import { ApiError } from "@/lib/api/error";
 import { getKeycardById, type KeycardResponse } from "@/lib/api/keycards";
 import {
   deriveVisitStatus,

--- a/app/(protected)/visits/page.tsx
+++ b/app/(protected)/visits/page.tsx
@@ -9,7 +9,7 @@ import FilterListIcon from "@mui/icons-material/FilterList";
 import RefreshIcon from "@mui/icons-material/Refresh";
 import VisibilityOutlinedIcon from "@mui/icons-material/VisibilityOutlined";
 import { Button } from "@/components/ui/button";
-import { ApiError } from "@/lib/apiClient";
+import { ApiError } from "@/lib/api/error";
 import {
   deriveVisitStatus,
   getVisits,

